### PR TITLE
collection_merge

### PR DESCRIPTION
**Pull request by brookeclouston**
Updated the original script to include try/except statement when deleting the empty collection, added more test methods to the test script.

### CODE REVIEW

**Comment by kcranston** [comment]
Thinking about long-term maintainability, and that this script makes significant database changes, I'd like to see better variable names (i.e. `source_collection` and `destination_collection` rather than `collection1` and `collection2`). Similarly, create variables with informative names rather than rather than referring directly to items in a list (`array[0]`). I'll leave some comments on the code, too.

File: CollectionMerge.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    # Returns list of catalog numbers in collection
+    db_catalog_numbers = database.cursor()
+    db_catalog_numbers.execute("SELECT CatalogNumber FROM collectionobject "
+                               "WHERE CollectionID = '%s'" % collectionid)
+    return [[catalog_number[0]] for catalog_number in db_catalog_numbers.fetchall()]
+
+def check_conflicts(database, collection1, collection2):
+    # Returns list of conflicting catalog numbers and list of catalog numbers from collection1
+    cat_numbers1 = fetch_catalog_numbers(database, collection1)
+    cat_numbers2 = fetch_catalog_numbers(database, collection2)
```
Comment (kcranston) [inline CollectionMerge.py]: also rename `cat_numbers` variables when you rename `collection1` and `collection2`

File: CollectionMerge.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        db_delete.execute("DELETE FROM collection WHERE collectionID = '%s'" % collectionid)
+        database.commit()
+        return None
+    except integrity_error:
+        return False
+
+def csv_report(file_name, headings, result_data, show):
+    # Writes report, displays data if specified
+    if show:
+        for row in result_data:
```
Comment (kcranston) [inline CollectionMerge.py:48]: would be helpful to print the header, too

File: CollectionMerge.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            switch_references(database, table_list, collection1, collection2)
+            print("%s collection objects merged" % len(catalog_numbers[1]))
+            if delete:
+                delete = delete_collection(database, collection1, integrity_error)
+                if delete:
+                    return print("Integrity Error occurred when attempting to delete collection "
+                                 "%s, see -h for more options" % collection1)
+                print("Collection %s deleted from schema" % collection1)
+            file_name = "CatalogNumbersMerged[%s]" % (datetime.date.today())
+            csv_report(file_name, headings, catalog_numbers[1], show)
```
Comment (kcranston) [inline CollectionMerge.py]: use variable instead of `catalog_numbers[1]`

File: CollectionMerge.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    show = args.show
+    merge = args.merge
+    delete = args.delete
+    integrity_error = pymysql.err.IntegrityError
+    try:
+        database = pymysql.connect("localhost", args.username, args.password, args.database)
+    except pymysql.err.OperationalError:
+        return print("Error connecting to database")
+    if report:
+        catalog_numbers = check_conflicts(database, collection1, collection2)
```
Comment (kcranston) [inline CollectionMerge.py]: a comment here about what gets returned from check_conflicts would be helpful

File: CollectionMerge.py (hunk 1 above)
Comment (brookeclouston) [reply CollectionMerge.py]: Variables have been renamed for clarity

File: CollectionMerge.py (hunk 2 above)
Comment (brookeclouston) [reply CollectionMerge.py:48]: The heading will now be printed with the catalog number displayed

File: CollectionMerge.py (hunk 3 above)
Comment (brookeclouston) [reply CollectionMerge.py]: Variable names have been updated for clarity

File: CollectionMerge.py (hunk 4 above)
Comment (brookeclouston) [reply CollectionMerge.py]: A comment has been added

**Comment by kcranston** [comment]
Awesome! Thanks! Much easier to follow, now.

**Pull request merged**